src/binder/assemblybinder.cpp

```diff
@@ -1158,13 +1165,48 @@ namespace BINDER_SPACE
                                                                 pBindResult));
             }
         }
-        else
+        else 
         {
-            // Is assembly on TPA list?
+            ReleaseHolder<Assembly> pTPAAssembly;
             SString &simpleName = pRequestedAssemblyName->GetSimpleName();
+
+            // Is assembly in the bundle?
+            // Single-file bundle contents take precedence over TPA.
+            // Bundled assemblies are treated similar to TPAs.
+            if (BundleInfo::AppIsBundle())
+            {
+                SString candidates[] = { W(".dll"), W(".ni.dll") };
+
+                // Loop through the binding paths looking for a matching assembly
+                for (DWORD i = 0; i < 2; i++)
+                {
+                    SString bundledName(simpleName);
+                    bundledName.Append(candidates[i]);
+
+                    hr = GetAssembly(bundledName,
+                        TRUE,  // fIsInGAC
+                        FALSE, // fExplicitBindToNativeImage
+                        &pTPAAssembly);
```

Comment:
This is sort of wrong - what we want here is to check the bundle probe - if it finds it, great - load that. But if the probe doesn't find it, we want to skip this step and instead continue below with the TPA search.
Right now I think this will actually behave differently:
* If the probe finds it - it will work fine
* If the probe doesn't find it, I think we will eventually call `CreateFile(assemblyFileName, ...)` - which I think will search for the file in current directory (didn't try this though) - we should avoid that behavior. Specifically if the assembly is still in TPA, we should use the path from TPA in this case, even if we can find the assembly also in current directory (which may not be the app's directory at all - so we might end up loading random code basically).

The short version is that this change modifies GetAssembly contract - before the path was assumed to be absolute (and it always was) - now it can be relative, but we don't limit that case to just probing the bundle.